pgjdbc/src/main/java/org/postgresql/jdbc/BooleanTypeUtil.java

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright (c) 2017, PostgreSQL Global Development Group
+ * See the LICENSE file in the project root for more information.
+ */
+
+package org.postgresql.jdbc;
+
+import org.postgresql.util.GT;
+import org.postgresql.util.PSQLException;
+import org.postgresql.util.PSQLState;
+
+/**
+ * Helper class to handle boolean type of PostgreSQL.
+ * <p>
+ * Based on values accepted by the PostgreSQL server:
+ * https://www.postgresql.org/docs/current/static/datatype-boolean.html
+ */
+class BooleanTypeUtil {
+
+  private BooleanTypeUtil() {
+  }
+
+  /**
+   * Cast an Object value to the corresponding boolean value.
+   *
+   * @param in Object to cast into boolean
+   * @return boolean value corresponding to the cast of the object
+   * @throws PSQLException PSQLState.INVALID_PARAMETER_TYPE
+   */
+  static boolean castToBoolean(final Object in) throws PSQLException {
+    if (in instanceof Boolean) {
+      return (Boolean) in;
+    }
+    if (in instanceof String) {
+      return from((String) in);
+    }
+    if (in instanceof Character) {
+      return from((Character) in);
+    }
+    if (in instanceof Number) {
+      return from((Number) in, in.getClass().getName());
+    }
+    throw cannotCastException(in.getClass().getName());
+  }
+
+  private static boolean from(final String strval) throws PSQLException {
+    // Leading or trailing whitespace is ignored, and case does not matter.
+    final String val = strval.trim();
+    if ("1".equals(val) || "true".equalsIgnoreCase(val)
+        || "t".equalsIgnoreCase(val) || "yes".equalsIgnoreCase(val)
+        || "y".equalsIgnoreCase(val) || "on".equalsIgnoreCase(val)) {
+      return true;
+    }
+    if ("0".equals(val) || "false".equalsIgnoreCase(val)
+        || "f".equalsIgnoreCase(val) || "no".equalsIgnoreCase(val)
+        || "n".equalsIgnoreCase(val) || "off".equalsIgnoreCase(val)) {
+      return false;
+    }
+    throw cannotCastException("String");
```

Comment:
I wonder if adding actual "bad" value to the error message would count as a security violation.
I think adding the value explicitly would simplify the debugging.

Any thoughts?